### PR TITLE
Improve interfaces

### PR DESCRIPTION
**Pull request by TolisChal**
1. Add C++ test for new volume algorithm and billiard walk.

2. Add zonotope, V- and H- random generators.

3. Improve both interfaces and documentation.

4. Fix bugs in both C++ and R interfaces.

5. Fix a bug in cooling_hpoly.h

### CODE REVIEW

File: test/generator.cpp [hunk 1]
```diff
@@ -207,22 +281,45 @@ int main(const int argc, const char** argv) {
...
     }
+    if (Hpoly && rand) kind = 6;
 
     if (Zono) {
         if (m > 0) {
-            Zonotope ZP = gen_zonotope<Zonotope, RNGType>(d, m);
-            create_txt(ZP.get_mat(), ZP.get_vec(), kind, true);
+            Zonotope ZP;
+            switch (kind) {
+                case -1:
```
Comment (vissarion) [inline test/generator.cpp]: is it possible to get -1 ? why not putting a default case?

File: test/generator.cpp [hunk 2]
```diff
@@ -29,14 +33,46 @@ void create_txt(MT A, VT b, int kind, bool Vpoly) {
     std::string name;
 
     std::ofstream outputFile;
-    if(Vpoly) {
-        if (kind == 0) {
+    if(Zono) {
```
[the remark this answers is not in the document]
Comment (TolisChal) [reply test/generator.cpp]: I put spaces everywhere.

File: test/generator.cpp [hunk 3]
```diff
@@ -29,14 +33,46 @@ void create_txt(MT A, VT b, int kind, bool Vpoly) {
...
+        if (kind == 1) {
```
[the remark this answers is not in the document]
Comment (TolisChal) [reply test/generator.cpp]: I reduced the code.

File: test/generator.cpp [hunk 4]
```diff
@@ -29,14 +33,46 @@ void create_txt(MT A, VT b, int kind, bool Vpoly) {
...
+            std::string poly = "zonotope";
+            name = poly + bar + std::to_string(d) + bar + std::to_string(m) + ext;
+            outputFile.open(name);
+            outputFile << name << "\n";
+            outputFile << "Zonotpe\n";
```
[the remark this answers is not in the document]
Comment (TolisChal) [reply test/generator.cpp]: thanks, done.

File: test/generator.cpp [hunk 5]
```diff
@@ -29,14 +33,46 @@ void create_txt(MT A, VT b, int kind, bool Vpoly) {
...
             name = poly + bar + std::to_string(d) + bar + std::to_string(m) + ext;
             outputFile.open(name);
-            outputFile<<name<<"\n";
-            outputFile<<"Zonotpe\n";
-        } else if (kind == 1) {
+            outputFile << name << "\n";
+            outputFile << "Zonotpe\n";
+        }
+    }else if(Vpoly) {
+        if (kind == 4) {
```
[the remark this answers is not in the document]
Comment (TolisChal) [reply test/generator.cpp]: done.

File: test/generator.cpp (hunk 1 above)
Comment (TolisChal) [reply test/generator.cpp]: I set a default choice.

**Pull request merged**